server_v2.py

```diff
@@ -229,17 +242,24 @@ def execute_stmt(conn, sqls, stmt):
     try:
         changes_before = conn.total_changes()
         prepared, sql_rest = conn.prepare(sql)
+        if not prepared:
+            raise ResponseError(f"Could not prepare statement: {sql}", "ARGS_INVALID")
```

Comment:
`prepared` is `None` when the input SQL text does not contain any SQL statements. The correct error code to return in this case is `SQL_NO_STATEMENT` and the message can be more concrete by saying something like "SQL string does not contain any statement" (as in sqld). The error message also should not include the SQL string (or at least include its `repr`, i.e. `{sql!r}`), because the client knows what SQL they sent and other statement-related errors also don't include the statement.

Reply:
changed to `SQL_NO_STATEMENT`, but it also fails if the statement contains a leading null byte (that's the python test suit does).